### ISSUE TITLE
get/set API for namespace policy need to be refactored

### ISSUE THREAD

**Issue by yush1ga** [apache/incubator-pulsar#1317]
Namespace policy has many properties and get/set API exists for each property.
If we add new property, we have to create get/set API for REST(v1, v2)/Java/CLI.
I think it is inconvenience for both developer and user.

Therefore, I think we should refactor get/set API for namespace policy.
For example, like following

```
# REST
GET /admin/namespaces/{property}/{cluster}/{namespace}/policies/{policyProperty}
POST /admin/namespaces/{property}/{cluster}/{namespace}/policies/{policyProperty}

# Java
admin.namespaces().getPolicy(policyProperty)
admin.namespaces().setPolicy(policyProperty, policyValue)

# CLI
pulsar-admin namespaces get-policies --property policyProperty
pulsar-admin namespaces set-policies --property policyProperty --value policyValue
```

What do you think ?

**Comment by sijie** [comment]
sounds like a good approach to take. +1

**Comment by merlimat** [comment]
I completely agree with the goal of simplification, but I think the problem is that each property is not of the same type: some are simple long/string scalars, while others are complex objects.

The validations performed also are kind of different for each property.

Finally, while the refactoring looks good for REST, I would still prefer to have individual operations on the Java/CLI since it makes the user life easier, by discovering all the possible methods/commands.

**Comment by maskit** [comment]
I like the proposed idea. 👍 

My thoughts:

1) Maintainability
Since we have many interfaces for policies (API for 3 languages, CLI, and REST), maintainability is unignorable. Replacing all individual interfaces with a simple get / set interface would absolutely improve maintainability of the code. And it would also be good for compatibility.

2) Approachability
People like code completion. From my experience, list of properties / methods are really useful especially when you use unfamiliar libraries. It tells you not only candidates, but also features you didn't know. To me, losing it would be a pain and documentation wouldn't be enough complement. However, I'd be happy if type of `policyProperty` in the proposal is `Enum` because I would be able to see the list. Of course we need to maintain the enum, but maintaining enum entries should be much easier than maintaining methods.

3) Data types
What need to be checked for a policy and the policy name should be 1:1, as far as I know. Looking up a `PolicyValidator` doesn't sound difficult to me.

Nitpick) REST API
http://restcookbook.com/HTTP%20Methods/put-vs-post/
https://developer.github.com/v3/#http-verbs

**Comment by sijie** [comment]
This is implemented by #1380
